lib/shopify_app/login_protection.rb

```diff
@@ -60,5 +60,52 @@ def login_url(params = {})
     rescue NoMethodError => e
       shopify_app.login_url(params)
     end
+
+    def redirect_to_with_fallback(url)
+      url_json = url.to_json
+
+      render inline: %Q(
+        <!DOCTYPE html>
+        <html lang="en">
+          <head>
+            <meta charset="utf-8" />
+            <meta http-equiv="refresh" content="0; url={url_json}>
```

Comment:
missing a `#` here, will add shortly (i'll also probably need to do wrap this in single quotes rather than double).